Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/SurveySettings.kt

```diff
@@ -1,78 +1,88 @@
 package de.rki.coronawarnapp.datadonation.survey
 
-import android.content.Context
-import com.google.gson.Gson
+import androidx.annotation.VisibleForTesting
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringPreferencesKey
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
 import de.rki.coronawarnapp.datadonation.OTPAuthorizationResult
 import de.rki.coronawarnapp.datadonation.OneTimePassword
-import de.rki.coronawarnapp.util.di.AppContext
-import de.rki.coronawarnapp.util.preferences.clearAndNotify
+import de.rki.coronawarnapp.util.datastore.clear
+import de.rki.coronawarnapp.util.datastore.dataRecovering
+import de.rki.coronawarnapp.util.datastore.distinctUntilChanged
 import de.rki.coronawarnapp.util.reset.Resettable
-import de.rki.coronawarnapp.util.serialization.BaseGson
+import de.rki.coronawarnapp.util.serialization.BaseJackson
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
 import timber.log.Timber
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
 class SurveySettings @Inject constructor(
-    @AppContext val context: Context,
-    @BaseGson val gson: Gson
+    @SurveySettingsDataStore private val dataStore: DataStore<Preferences>,
+    @BaseJackson private val objectMapper: ObjectMapper,
 ) : Resettable {
 
-    private val preferences by lazy {
-        context.getSharedPreferences("survey_localdata", Context.MODE_PRIVATE)
+    val oneTimePassword = dataStore.dataRecovering.distinctUntilChanged(
+        key = KEY_OTP
+    ).map { json ->
+        if (json != null && json != String()) {
```

Comment:
`!json.isNullOrEmpty()`

Reply:
Made nullable, so empty check is removed